pkgs/development/python-modules/skops/default.nix

```
@@ -0,0 +1,103 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+
+  pythonOlder,
+
+  pytestCheckHook,
+  pythonRelaxDepsHook,
+
+  setuptools,
+
+  # for tests
+  rich,
+  catboost,
+  xgboost,
+  lightgbm,
+  pandas,
+  matplotlib,
+  flaky,
+  quantile-forest,
+
+  packaging,
+  tabulate,
+  huggingface-hub,
+  scikit-learn,
+}:
+
+buildPythonPackage rec {
+  pname = "skops";
+  version = "0.9.0";
+  pyproject = true;
+
+  disabled = pythonOlder "3.8";
+
+  src = fetchFromGitHub {
+    owner = "skops-dev";
+    repo = "skops";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-JCK5x2OKRcUftld+bGgHD74eB3Jmij2PZnOPIenTRlM=";
+  };
+
+  # pyproject contains only dev settings
+  # and pytest config with no need coverage options
+  postPatch = ''
+    rm pyproject.toml
+  '';
+
+  build-system = [ setuptools ];
+
+  dependencies = [
+    packaging
+    tabulate
+    huggingface-hub
+    scikit-learn
+  ];
+
+  nativeCheckInputs = [
+    pytestCheckHook
+    pythonRelaxDepsHook
+
+    rich
+    catboost
+    xgboost
+    lightgbm
+    pandas
+    matplotlib
+    flaky
+    quantile-forest
+  ];
```

```suggestion
  optional-dependencies = {
    rich = [ rich ];
  };

  nativeCheckInputs = [
    pytestCheckHook
    pythonRelaxDepsHook

    catboost
    xgboost
    lightgbm
    pandas
    matplotlib
    flaky
    quantile-forest
  ] ++ optional-dependencies.rich;
```
https://github.com/skops-dev/skops/blob/9b565a0672a8c25c8eac38604a07cdcf7047f757/setup.py#L78-L82